src/lib.rs

```diff
@@ -71,29 +73,43 @@ impl ElectionResults {
 ///
 /// `Election` is constructed with a set of candidates and votes, and is consumed when it returns
 /// the results of the election.
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct Election {
     candidates: Vec<Candidate>,
     elected: CandidateVotesMap,
     eliminated: CandidateVotesMap,
     num_spoiled_votes: u64,
     seats: u64,
     votes: Vec<Vote>,
+    logger: slog::Logger,
 }
 
 impl Election {
     /// Manually construct an `Election` where the input data is already in memory.
     ///
     /// The more common way to construct an `Election` is with [`Election::from_csv_file`].
-    pub fn new(candidates: Vec<Candidate>, votes: Vec<Vote>, seats: u64) -> Result<Self, Error> {
+    pub fn new<L: Into<Option<slog::Logger>>>(
+        candidates: Vec<Candidate>,
+        votes: Vec<Vote>,
+        seats: u64,
+        logger: L,
+    ) -> Result<Self, Error> {
         let mut election = Election {
             candidates,
             votes,
             seats,
-            ..Default::default()
+            logger: logger
+                .into()
+                .unwrap_or_else(|| slog::Logger::root(slog_stdlog::StdLog.fuse(), o!())),
+            elected: Default::default(),
```

Comment:
why not use `..Default::default()` here ?

Reply:
That's because that syntax can only be used when `Election` derives `Default`, and right now it cannot because `Logger` doesn't and that is one of its fields.